src/node.h

```diff
@@ -722,7 +722,8 @@ NODE_EXTERN ArrayBufferAllocator* GetArrayBufferAllocator(IsolateData* data);
 // a snapshot and have a main context that was read from that snapshot.
 NODE_EXTERN v8::Local<v8::Context> GetMainContext(Environment* env);
 
-NODE_EXTERN void OnFatalError(const char* location, const char* message);
+[[noreturn]] NODE_EXTERN void OnFatalError(const char* location,
+                                           const char* message);
```

Comment:
Is this change ABI-compatible?

Reply:
I think so. The type of the function remains unchanged (i.e., can still be assigned to exactly the same function pointer types) and C++ name mangling is not affected by `[[noreturn]]`. The compiler may optimize both at call sites and within the function itself based on this attribute, but that should not matter for ABI.

Reply:
Yeah, `[[noreturn]]` doesn't change the name mangling and the type of the function.